Mathlib/Lean/Expr/Basic.lean

```diff
@@ -5,9 +5,7 @@ Authors: Mario Carneiro, Simon Hudon, Kim Morrison, Keeley Hoek, Robert Y. Lewis
 Floris van Doorn, Edward Ayers, Arthur Paulino
 -/
 import Mathlib.Init
-import Lean.Meta.Tactic.Rewrite
-import Batteries.Tactic.Alias
-import Lean.Elab.Binders
+import Lean.Expr
```

Comment:
Is it worth enforcing this with an `assert_not_imported` here?

Reply:
Good question! My natural answer is rather "should we make sure `Mathlib/Lean` never imports batteries? (At the moment Mathlib/Lean/Expr/Rat imports Batteries.Tactic.Alias, but spuriously at best... was that a conscious decision at some point? In any point, it may be nicer to systematically discuss in a follow-up.)